pkg/collector/corechecks/cluster/orchestrator/collectors/ecs/task.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors/ecs"
+	transformers "github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/transformers/ecs"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// TaskCollector is a collector for ECS tasks.
+type TaskCollector struct {
+	metadata  *collectors.CollectorMetadata
+	processor *processors.Processor
+}
+
+// NewTaskCollector creates a new collector for the ECS Task resource.
+func NewTaskCollector() *TaskCollector {
+	return &TaskCollector{
+		metadata: &collectors.CollectorMetadata{
+			IsStable:           false,
+			IsMetadataProducer: true,
+			IsManifestProducer: false,
+			Name:               "ecstasks",
+			NodeType:           orchestrator.ECSTask,
+		},
+		processor: processors.NewProcessor(new(ecs.TaskHandlers)),
+	}
+}
+
+// Metadata is used to access information about the collector.
+func (t *TaskCollector) Metadata() *collectors.CollectorMetadata {
+	return t.metadata
+}
+
+// Init is used to initialize the collector.
+func (t *TaskCollector) Init(_ *collectors.CollectorRunConfig) {}
+
+// Run triggers the collection process.
+func (t *TaskCollector) Run(rcfg *collectors.CollectorRunConfig) (*collectors.CollectorRunResult, error) {
+	list := rcfg.WorkloadmetaStore.ListECSTasks()
+	tasks := make([]transformers.TaskWithContainers, 0, len(list))
+	for _, task := range list {
+		tasks = append(tasks, t.fetchContainers(rcfg, task))
+	}
+
+	ctx := &processors.ECSProcessorContext{
+		BaseProcessorContext: processors.BaseProcessorContext{
+			Cfg:              rcfg.Config,
+			MsgGroupID:       rcfg.MsgGroupRef.Inc(),
+			NodeType:         t.metadata.NodeType,
+			ManifestProducer: t.metadata.IsManifestProducer,
+		},
+	}
+	if len(list) > 0 {
+		ctx.AWSAccountID = list[0].AWSAccountID
+		ctx.ClusterName = list[0].ClusterName
+		ctx.Region = list[0].Region
+
+		// If the cluster ID is not set, we generate it from the first task
+		if rcfg.ClusterID == "" {
+			rcfg.ClusterID = initClusterID(list[0].AWSAccountID, list[0].Region, list[0].ClusterName)
+		}
+		ctx.ClusterID = rcfg.ClusterID
+	}
+
+	processResult, processed := t.processor.Process(ctx, tasks)
+
+	if processed == -1 {
+		return nil, fmt.Errorf("unable to process resources: a panic occurred")
+	}
+
+	result := &collectors.CollectorRunResult{
+		Result:             processResult,
+		ResourcesListed:    len(list),
+		ResourcesProcessed: processed,
+	}
+
+	return result, nil
+}
+
+// fetchContainers fetches the containers from workloadmeta store for a given task.
+func (t *TaskCollector) fetchContainers(rcfg *collectors.CollectorRunConfig, task *workloadmeta.ECSTask) transformers.TaskWithContainers {
+	ecsTask := transformers.TaskWithContainers{
+		Task:       task,
```

Comment:
will task here be reset to the last value of task in the parent loop, for all entries? The ptr is reused in the for.. I'm not sure if that cascades into this function call.

Reply:
It won't but I could never figure out this looping variable issue, so I updated it to make sure it's always correct, and updated the integration test to include this part
https://github.com/DataDog/datadog-agent/pull/22060/commits/9fd8b7a3d7ac7cdf81ad97a36cf575b91067344e